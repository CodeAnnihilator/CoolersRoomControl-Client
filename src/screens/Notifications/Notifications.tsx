import React from 'react';

import NotificationsTemplate from '../../common/components/NotificationsTemplate/NotificationsTemplate';

interface INotification {
	id: number;
	type: string;
	date: string;
	time: string;
	operation?: string;
	temperature?: string;
	humidity?: string;
	weekly?: boolean;
	description?: string;
	message?: string;
	model?: string;
	serial?: string;
	roomId: number;
}

interface IProps {
	dates: number[];
	notifications: INotification[][];
}

const Notifications: React.FC<IProps> = ({dates, notifications}) => (
	<NotificationsTemplate dates={dates} notifications={notifications} />
);

export default Notifications;
