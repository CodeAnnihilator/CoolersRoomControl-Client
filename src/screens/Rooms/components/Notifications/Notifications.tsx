import React from 'react';

import NotificationsTemplate from '../../../../common/components/NotificationsTemplate/NotificationsTemplate';

interface INotification {
	id: number;
	type: string;
	date: string;
	time: string;
	message?: string;
	model?: string;
	serial?: string;
	roomId: number;
}

interface IProps {
	dates: number[];
	notifications: INotification[][];
}

const Notifications: React.FC<IProps> = ({dates, notifications}) => (
	<NotificationsTemplate dates={dates} notifications={notifications} />
);

export default Notifications;
