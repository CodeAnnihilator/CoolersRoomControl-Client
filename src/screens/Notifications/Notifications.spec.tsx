import * as React from 'react';
import renderer from 'react-test-renderer';

import Notifications from '../Notifications/Notifications';

describe('Notifications', () => {
	it('+++capturing Snapshot of Notifications', () => {
		const dates = [111, 222];
		const notifications = [
			[{
				id: 1,
				type: 'type',
				date: 'date',
				time: 'time',
				operation: 'operation',
				temperature: 'temperature',
				humidity: 'humidity',
				weekly: true,
				description: 'description',
				message: 'message',
				model: 'model',
				serial: 'serial',
				roomId: 1,
			}],
			[{
				id: 1,
				type: 'type',
				date: 'date',
				time: 'time',
				operation: 'operation',
				temperature: 'temperature',
				humidity: 'humidity',
				weekly: true,
				description: 'description',
				message: 'message',
				model: 'model',
				serial: 'serial',
				roomId: 1,
			}],
		];
		const renderedValue =  renderer.create(<Notifications dates={dates} notifications={notifications} />).toJSON();

		expect(renderedValue).toMatchSnapshot();
});
});
