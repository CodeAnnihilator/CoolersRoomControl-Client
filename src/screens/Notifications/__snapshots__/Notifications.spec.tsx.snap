// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications +++capturing Snapshot of Notifications 1`] = `null`;
